File: android/Comandas/app/src/main/java/com/valleapp/comandas/tareas/TareaManejarAutorias.java
package com.valleapp.comandas.tareas;

import android.content.ContentValues;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.support.annotation.NonNull;
import android.util.Log;

import com.valleapp.comandas.utilidades.HTTPRequest;

import org.json.JSONArray;
import java.util.TimerTask;


public class TareaManejarAutorias extends TimerTask {

    Handler mainHandler;
    String idautorizado;
    int autorizacionesSize = 0;
    int timeOut;
    boolean procesado = true;
    String url;

    public  TareaManejarAutorias(Handler mainHandler, String idautorizado, int timeOut, String url){
        this.timeOut = timeOut; this.idautorizado = idautorizado; this.mainHandler= mainHandler;
        this.url = url;
    }


    final  Handler  handler = new  Handler(Looper.getMainLooper()){
        @Override
        public void handleMessage(@NonNull Message msg) {
            try {
                String res = msg.getData().getString("RESPONSE");
                if(res != null && res != "") {
                    JSONArray l = new JSONArray(res);
                    if (l.length() != autorizacionesSize) {
                        autorizacionesSize = l.length();
                        if (mainHandler != null) mainHandler.handleMessage(msg);
                    }
                }
            }catch (Exception e){
                Log.w("Autorias", "Respuesta erronea del servidor");
            }
            procesado = true;
        }
    };

    @Override
    public void run() {
            if (procesado){
                procesado = false;
                ContentValues p = new ContentValues();
                p.put("idautorizado", idautorizado);
                new HTTPRequest(url, p, "autorias", handler);
            }
            synchronized (this){
                try {
                    wait(timeOut);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
    }
}

File: android/ValleTPV/ValleCOM/src/main/java/com/valleapp/vallecom/tareas/TareaManejarAutorias.kt
package com.valleapp.vallecom.tareas

import android.content.ContentValues
import android.os.Handler
import android.os.Looper
import android.os.Message
import android.util.Log
import com.valleapp.valletpvlib.comunicacion.HTTPRequest
import org.json.JSONArray
import java.util.TimerTask

class TareaManejarAutorias(
    private val mainHandler: Handler?,
    private val idautorizado: String,
    private val timeOut: Int,
    private val url: String,
    private val uid: String
) : TimerTask() {

    private var autorizacionesSize = 0
    private var procesado = true

    private val handler = object : Handler(Looper.getMainLooper()) {
        override fun handleMessage(msg: Message) {
            try {
                val res = msg.data.getString("RESPONSE")
                if (!res.isNullOrEmpty()) {
                    val l = JSONArray(res)
                    if (l.length() != autorizacionesSize) {
                        autorizacionesSize = l.length()
                        mainHandler?.handleMessage(msg)
                    }
                }
            } catch (e: Exception) {
                Log.w("Autorias", "Respuesta erronea del servidor")
            }
            procesado = true
        }
    }

    override fun run() {
        if (procesado) {
            procesado = false
            val p = ContentValues().apply {
                put("idautorizado", idautorizado)
                put("uid", uid)
            }
            HTTPRequest(url, p, "autorias", handler)
        }
        synchronized(this) {
            try {
                (this as Object).wait(timeOut.toLong())
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }
    }
}
